core/node.go

```diff
@@ -143,9 +198,51 @@ func (n *Node) ID() peer.ID {
 // Start causes the Node to continuously send messages to and receive messages
 // from its peers. It blocks until an error is encountered or `Stop` is called.
 func (n *Node) Start() error {
+	// Note: dht.Bootstrap has a somewhat confusing name. It doesn't automatically
+	// connect to the bootstrap peers. It just starts the background process of
+	// searching for new peers.
+	if err := n.dht.Bootstrap(n.ctx); err != nil {
+		return err
+	}
+
+	// Advertise ourselves for the purposes of peer discovery.
+	discovery.Advertise(n.ctx, n.routingDiscovery, n.config.RendezvousString, discovery.TTL(advertiseTTL))
```

Comment:
At some point we'll probably want to add a persistent advertisement, such that libp2p manages the re-advertisement when the TTL expires.

Reply:
Yes, the short TTL is just for testing purposes right now. We're generating a random peer ID each time you run *main.go* so with a longer TTL we would see a lot of attempts to connect to a peer that disconnected a long time ago.